src/utils/utils_posix_common.c

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+int util_env_var(const char *envvar, char *buffer, size_t buffer_size) {
+    char *value = getenv(envvar);
+    if (!value) {
+        return 0;
+    }
+
+    size_t len = strlen(value) + 1;
+    if (len > buffer_size) {
+        return len;
```

Comment:
isn't better to return negative value here? or even better return "buffer_size" as a negative value so the caller knows how big buffer is needed.

Reply:
It will also make return value parse simpler  instead 
`int rc = util_env_var(envvar, buffer, BUFFER_SIZE);
    if (0 < rc && rc < BUFFER_SIZE) {`
  you can just do 
`if (util_env_var(envvar, buffer, BUFFER_SIZE) > 0) {`

Reply:
Done.